tests/sles4sap/ensa/netweaver_ensa2_cluster.pm

```diff
@@ -17,8 +17,15 @@ use lockapi;
 sub run {
     my ($self) = @_;
     # Set maintenance to 'true'
-    assert_script_run("crm configure property maintenance-mode='true'");
     my $instance_type = get_required_var('INSTANCE_TYPE');
+    my $hosts_shared_dir;
+    my $usr_sap_device_ascs;
+    my $usr_sap_device_ers;
+    if ($instance_type eq 'ASCS') {
+        $hosts_shared_dir = get_required_var('HOSTS_SHARED_DIRECTORY');
```

Comment:
Why does this need to be here and not the following if clause? What does $usr_sap_device_acss gives us that just using `get_required_var` while building the hash doesnt?

Reply:
I think if test module has to fail just for missing setting it is better that it fails for this as soon as possible even before starting to touch the SUT

Reply:
Why do you think that calling `get_required_variable` and eventually `die` is better where it was, as part of `%template_variables` definitions?

Reply:
The required vars will only be there if the schedule is wrong. These values are set by  ensa_filesystems.pm  if the test gets that far it is already late. This is a `get_required_var` not because there is a real risk we will not have the `var`, but because we need to show that we absolutely need them to have a coherent test.

Reply:
I'm with @a-kpappas here: I don't see the need for the new variables when the previous version was already assigning these inside a HASH (which is basically a different type of variable). The other thing with the change is that it's making the module reserve memory space for 2 extra variables (yeah, I know they don't cost that much) whether the module runs on an ASCS node or in an ERS node ... the previous version only creates the variable in ASCS node, which means it wastes marginally less memory. Other than that, changes LGTM.